cmd/openfga/openfga_tls_test.go

```diff
@@ -0,0 +1,208 @@
+)
+
+const (
+	tlsEnableEnvVar   = "OPENFGA_ENABLE_TLS"
+	tlsCertPathEnvVar = "OPENFGA_TLS_CERT_PATH"
+	tlsKeyPathEnvVar  = "OPENFGA_TLS_KEY_PATH"
+
+	caCert = `-----BEGIN CERTIFICATE-----
+MIIDpzCCAo+gAwIBAgIUayZ6IiyldzKDZIA42b1ByVu4Hd8wDQYJKoZIhvcNAQEL
+BQAwWjELMAkGA1UEBhMCSlAxEDAOBgNVBAgTB0Z1a3Vva2ExEDAOBgNVBAcTB0Z1
+a3Vva2ExEzARBgNVBAoTCmpvYi13b3JrZXIxEjAQBgNVBAMTCWxvY2FsaG9zdDAe
+Fw0yMjA2MDYyMzQzMDBaFw0yNzA2MDUyMzQzMDBaMFoxCzAJBgNVBAYTAkpQMRAw
+DgYDVQQIEwdGdWt1b2thMRAwDgYDVQQHEwdGdWt1b2thMRMwEQYDVQQKEwpqb2It
+d29ya2VyMRIwEAYDVQQDEwlsb2NhbGhvc3QwggEiMA0GCSqGSIb3DQEBAQUAA4IB
+DwAwggEKAoIBAQDCgzzKv71AHNqdWA12wDlWY2zdlQoW0Flt+GyD9itEincSdllW
+Bw9m5RvTZjScAZS0/veJ/ARttmvRRUcFTiL88SZl68heVXQbzGM2ks3mcnpJdg3T
+Alq61h4gqSewYzgN9UTx7ftCc5ga5DEuzu8Sq//KKzSh08/7/ToXVYWxyiW1MQMh
+DRB+l9OyDYP9sIQG0kiMgcSfsUOmy3BON3dILZ1W4Y7kVK9K4ES03LkAi98daIHb
+MkBtJIbgijoOeOTf6R1zaS8vdqXvDvyqtg/lYOf2UdxwAoHhhYerA3+hzAnhptb/
+WF3fKB3Yv7iXNCP00EhEvPczLrKkDYoErXD3AgMBAAGjZTBjMA4GA1UdDwEB/wQE
+AwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBTncJnA3mUMIQaMmzdgnZBI
++19t3zAhBgNVHREEGjAYhwR/AAABhxAAAAAAAAAAAAAAAAAAAAABMA0GCSqGSIb3
+DQEBCwUAA4IBAQCeTdZwmfMhxG8HT7JipFjy5ZMM0Aj1qbdToy7FBuyJpLUuyMSp
+pGkkpnvTcvEDtKsPtBY/tBTVRSq6y4bq8Wz1LC0qUy2bKXF1LnUv1DwSUC0Vl4IH
+rgqsK8SGqVCXtTpSOtZNwi2hHsQy5r2cHaRXGa3D32qCav8HTMJg1VXphRq++QF/
+AUcbaAeT+lg1swidlYf/ZnAlp8QE/pbbCbA0K2Kj8DQrKKeVPsILLOWZac66nhGm
+GPUWzKp4A18yOcOaCL7XkhcWjIuIrTmMrnwhupSpCTJfPQ1yCtBHJuC72sOMt1Ps
+5MMtvl22bRz2B/wGwcgsJ0tv/PkV5YDJY2MG
+-----END CERTIFICATE-----`
+
+	serverCert = `-----BEGIN CERTIFICATE-----
+MIID0DCCArigAwIBAgIUMVe743fpJExbqklNX4ln2UOZ31wwDQYJKoZIhvcNAQEL
+BQAwWjELMAkGA1UEBhMCSlAxEDAOBgNVBAgTB0Z1a3Vva2ExEDAOBgNVBAcTB0Z1
+a3Vva2ExEzARBgNVBAoTCmpvYi13b3JrZXIxEjAQBgNVBAMTCWxvY2FsaG9zdDAe
+Fw0yMjA2MDYyMzQzMDBaFw0yMzA2MDYyMzQzMDBaMFoxCzAJBgNVBAYTAkpQMRAw
+DgYDVQQIEwdGdWt1b2thMRAwDgYDVQQHEwdGdWt1b2thMRMwEQYDVQQKEwpqb2It
+d29ya2VyMRIwEAYDVQQDEwlsb2NhbGhvc3QwggEiMA0GCSqGSIb3DQEBAQUAA4IB
+DwAwggEKAoIBAQCr5BWTiPVu4hUlA1CsbA+wdMifQvmSlzuxDTel9lUyqnGzozxM
+9Qi3DgsBKxLFyDlsEuBDe//xt73DXOrscu6qvddLju7jjggl4Xcr+gcd20ZdqRbL
+79Mgidaq45FtZCmPgIYGZZ0PPC/YoUGEJHf/RFtXzBemeTu+aqaHZukqHflhOoI9
+zH4wEth5Of31BmIr2GLIfJNyNPor7yMnAI++3CrsmS+lY5W8rqi++NFh70qz5meH
+xVmslRtGZRCRtAqMXxZUnYwB/4YZI94B4EOxRsFotbKGXvJjBiGn08TQssx+trux
+ln7gLCFIcW9pEfNZ9DhblbRtZjZqHOKfxs87AgMBAAGjgY0wgYowDgYDVR0PAQH/
+BAQDAgWgMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjAMBgNVHRMBAf8E
+AjAAMB0GA1UdDgQWBBSSa2txmZzKI7OBeCHIiTDMEXAu3DAsBgNVHREEJTAjggls
+b2NhbGhvc3SHBH8AAAGHEAAAAAAAAAAAAAAAAAAAAAEwDQYJKoZIhvcNAQELBQAD
+ggEBAG8e8Ga2wUlk6UUN5sg4dooPZgnAipHjkj8AYud9AExNf4o+2KOmA8EqQyYs
+r8WmgDSB853osJ1RyrApAedsXj4V1vgqZmn6XBUbKupGWsIEUSQ/+BcviSICFQNM
+NKe3yRA7FUrTVL+fou8QHtYyYIswqi75+TcgO3SKbmhHPU6I9zecPCYQAqEWvOP9
+zhAaWrtwr07/2nt7eED+B6qNNsYRu1/A9qSDMhPK69MZ6ZumF97ypmgvi7M34yCK
+EUnpHGvoDPbj4ZbaqXPTSC7DX4btrkYEHVy+5P+tKyoht7bPc8gMTxjQwVARNFnO
+iDLb3/R0pIqILWhmcG6xg0ymD4Y=
+-----END CERTIFICATE-----`
+
+	serverKey = `-----BEGIN RSA PRIVATE KEY-----
```

Comment:
not a huge fan of setting these keys like this.. is there no way of generating them at runtime?

Reply:
I agree it'd be nice to gen these in test setup.

We should be able to use 

`"crypto/rsa" # GenerateKey`
https://pkg.go.dev/crypto/rsa#GenerateKey


`"crypto/x509" # CreateCertificate`
https://pkg.go.dev/crypto/x509#CreateCertificate

Reply:
If you don't mind, can I do this in a future PR. It seems slightly involved.

Reply:
Task to track it: https://auth0team.atlassian.net/browse/IAMFGA-591